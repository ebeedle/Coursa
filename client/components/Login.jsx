import React from 'react';
import Heading from './Heading.jsx';
import NavBar from './NavBar.jsx';
import SingleInput from './SingleInput.jsx';
import $ from 'jquery'
import { BrowserRouter as Route, Redirect} from 'react-router-dom'


class Login extends React.Component {
  constructor() {
    super();
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.state = {
      username: '',
      password: '',
      redirect: false
    };
  }

  handleInputChange(event) {
    console.log('this.state.password :', this.state.password);
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;
    console.log('name :', name, 'value :', value);
    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    // alert('Your favorite flavor is: ' + this.state.value);
    //make post request to /login
    //if authenticated
    console.log('trying to log in')
    event.preventDefault();
    $.post('/login', { username: this.state.username, password: this.state.password })
    .done(() => {
      //redirect to home component
      this.setState({redirect: true}) 
    })
    .fail(e => {
      console.log('fail')
      alert('error try again :', e)
    })
  }

  render() {
    if (this.state.redirect) {
      return <Redirect to="/home" />;
    }
    return (
      <div>
        <NavBar additions={['', 'signup']}/>
        <div className="container padding-top">
          <div className="sign_up"> 
            Log In
          </div>
          <form onSubmit={this.handleSubmit}>
            <SingleInput
              title={"Email Address"}
              name={"username"}
              type={"email"}
              value={this.state.email}
              onChange={this.handleInputChange} />
            <SingleInput
              title={"Password"}
              name={"password"}
              type={"password"}
              value={this.state.password}
              onChange={this.handleInputChange} />
            <button type="submit" className="btn btn-default"> Submit </button>
          </form>
        </div>
      </div>
    )
  }
}

export default Login;