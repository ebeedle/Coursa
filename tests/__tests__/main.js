import React from 'react';
import { shallow } from 'enzyme';
import Login from '../../client/components/Login.jsx';

describe('Login', () => {
  it('should be defined', () => {
   expect(Login).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(
     <Login name='button test' />
    );
    expect(tree).toMatchSnapshot();
  });
});