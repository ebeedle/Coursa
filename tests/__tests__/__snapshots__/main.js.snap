// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          additions={
            Array [
              "",
              "signup",
            ]
          }
        />,
        <div
          className="container padding-top"
        >
          <div
            className="sign_up"
          >
            Log In
          </div>
          <form
            onSubmit={[Function]}
          >
            <SingleInput
              name="username"
              onChange={[Function]}
              title="Email Address"
              type="email"
              value={undefined}
            />
            <SingleInput
              name="password"
              onChange={[Function]}
              title="Password"
              type="password"
              value=""
            />
            <button
              className="btn btn-default"
              type="submit"
            >
               Submit 
            </button>
          </form>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additions": Array [
            "",
            "signup",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sign_up"
            >
              Log In
            </div>,
            <form
              onSubmit={[Function]}
            >
              <SingleInput
                name="username"
                onChange={[Function]}
                title="Email Address"
                type="email"
                value={undefined}
              />
              <SingleInput
                name="password"
                onChange={[Function]}
                title="Password"
                type="password"
                value=""
              />
              <button
                className="btn btn-default"
                type="submit"
              >
                 Submit 
              </button>
            </form>,
          ],
          "className": "container padding-top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "sign_up",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <SingleInput
                  name="username"
                  onChange={[Function]}
                  title="Email Address"
                  type="email"
                  value={undefined}
                />,
                <SingleInput
                  name="password"
                  onChange={[Function]}
                  title="Password"
                  type="password"
                  value=""
                />,
                <button
                  className="btn btn-default"
                  type="submit"
                >
                   Submit 
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "username",
                  "onChange": [Function],
                  "title": "Email Address",
                  "type": "email",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "title": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Submit ",
                  "className": "btn btn-default",
                  "type": "submit",
                },
                "ref": null,
                "rendered": " Submit ",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            additions={
              Array [
                "",
                "signup",
              ]
            }
          />,
          <div
            className="container padding-top"
          >
            <div
              className="sign_up"
            >
              Log In
            </div>
            <form
              onSubmit={[Function]}
            >
              <SingleInput
                name="username"
                onChange={[Function]}
                title="Email Address"
                type="email"
                value={undefined}
              />
              <SingleInput
                name="password"
                onChange={[Function]}
                title="Password"
                type="password"
                value=""
              />
              <button
                className="btn btn-default"
                type="submit"
              >
                 Submit 
              </button>
            </form>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "additions": Array [
              "",
              "signup",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="sign_up"
              >
                Log In
              </div>,
              <form
                onSubmit={[Function]}
              >
                <SingleInput
                  name="username"
                  onChange={[Function]}
                  title="Email Address"
                  type="email"
                  value={undefined}
                />
                <SingleInput
                  name="password"
                  onChange={[Function]}
                  title="Password"
                  type="password"
                  value=""
                />
                <button
                  className="btn btn-default"
                  type="submit"
                >
                   Submit 
                </button>
              </form>,
            ],
            "className": "container padding-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "sign_up",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <SingleInput
                    name="username"
                    onChange={[Function]}
                    title="Email Address"
                    type="email"
                    value={undefined}
                  />,
                  <SingleInput
                    name="password"
                    onChange={[Function]}
                    title="Password"
                    type="password"
                    value=""
                  />,
                  <button
                    className="btn btn-default"
                    type="submit"
                  >
                     Submit 
                  </button>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "username",
                    "onChange": [Function],
                    "title": "Email Address",
                    "type": "email",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "password",
                    "onChange": [Function],
                    "title": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Submit ",
                    "className": "btn btn-default",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": " Submit ",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
